python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+from hive_metastore.ttypes import (
+    FieldSchema,
+    InvalidOperationException,
+    MetaException,
+    NoSuchObjectException,
+    SerDeInfo,
+    StorageDescriptor,
+)
+from hive_metastore.ttypes import Table as HiveTable
+from thrift.protocol import TBinaryProtocol
+from thrift.transport import TSocket, TTransport
+
+from pyiceberg.catalog import Identifier, Properties
+from pyiceberg.catalog.base import Catalog
+from pyiceberg.exceptions import (
+    AlreadyExistsError,
+    NamespaceNotEmptyError,
+    NoSuchNamespaceError,
+    NoSuchTableError,
+)
+from pyiceberg.schema import Schema
+from pyiceberg.table.base import Table
+from pyiceberg.table.partitioning import PartitionSpec
+from pyiceberg.types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IcebergType,
+    IntegerType,
+    ListType,
+    LongType,
+    MapType,
+    StringType,
+    StructType,
+    TimestampType,
+    TimeType,
+    UUIDType,
+)
+
+hive_types = {
+    BooleanType: "boolean",
+    IntegerType: "int",
+    LongType: "bigint",
+    FloatType: "float",
+    DoubleType: "double",
+    DateType: "date",
+    TimeType: "string",
+    TimestampType: "timestamp",
+    StringType: "string",
+    UUIDType: "string",
+    BinaryType: "binary",
+    FixedType: "binary",
+    DecimalType: None,
+    StructType: None,
+    ListType: None,
+    MapType: None,
+}
+
+
+class _HiveClient:
+    _transport: TTransport
+    _client: Client
+
+    def __init__(self, url: str):
+        url_parts = urlparse(url)
+        transport = TSocket.TSocket(url_parts.hostname, url_parts.port)
+        self._transport = TTransport.TBufferedTransport(transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(transport)
+
+        self._client = Client(protocol)
+
+    def __enter__(self) -> Client:
+        self._transport.open()
+        return self._client
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._transport.close()
+
+
+class HiveCatalog(Catalog):
+    _client: _HiveClient
+
+    def __init__(self, name: str, properties: Properties, url: str):
+        super().__init__(name, properties)
+        self._client = _HiveClient(url)
+
+    def _storage_descriptor(self, schema: Schema, location: Optional[str]) -> StorageDescriptor:
+        ser_de_info = SerDeInfo(serializationLib="org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")
+        return StorageDescriptor(
+            self._columns(schema),
+            location,
+            "org.apache.hadoop.mapred.FileInputFormat",
+            "org.apache.hadoop.mapred.FileOutputFormat",
+            serdeInfo=ser_de_info,
+        )
+
+    def _columns(self, schema: Schema) -> List[FieldSchema]:
+        return [FieldSchema(field.name, self._convert_hive_type(field.field_type), field.doc) for field in schema.fields]
+
+    def _convert_hive_type(self, col_type: IcebergType) -> str:
```

Comment:
I think we'll want a real implementation for this using a type visitor, since the map is very basic.

Reply:
Since this is part of the write path, I'd postpone this until then. The visitor will be simple, but we need to test is thoroughly :)